src/main/scala/fi/oph/koski/valpas/log/AuditLog.scala

```diff
@@ -0,0 +1,44 @@
+package fi.oph.koski.valpas.log
+
+import fi.oph.koski.log.AuditLogMessage
+import fi.oph.koski.valpas.log.ValpasMessageField.ValpasMessageField
+import fi.oph.koski.valpas.log.ValpasOperation.ValpasOperation
+import fi.oph.koski.valpas.valpasuser.ValpasSession
+import fi.vm.sade.auditlog.{Changes, Operation, Target, User}
+import org.ietf.jgss.Oid
+
+object ValpasAuditLogMessage {
+  def apply(
+    operation: ValpasOperation,
+    extraFields: Map[ValpasMessageField, String]
+  )(implicit session: ValpasSession): AuditLogMessage = {
+    val target = extraFields.foldLeft(new Target.Builder()) { case (builder, (name, value)) =>
+      builder.setField(name.toString, value)
+    }.build
+    AuditLogMessage(
+      user = new User(
+        new Oid(session.user.oid),
+        session.clientIp,
+        session.user.serviceTicket.getOrElse(""),
+        session.userAgent),
+      operation = new ValpasAuditLogOperation(operation),
+      target = target,
+      changes = new Changes.Builder().build
+    )
+  }
+}
+
+object ValpasMessageField extends Enumeration {
+  type ValpasMessageField = Value
+  val OPPIJA_OID = Value
+}
+
+object ValpasOperation extends Enumeration {
+  type ValpasOperation = Value
+  val OPPIJA_KATSOMINEN = Value
```

Comment:
Mietin, että pitäisikö tässä olla nimenä ennemmin "VALPAS_OPPIJA_KATSOMINEN" tms.? Jos nämä kuitenkin menevät koski-backendin logille, ja sieltä jollain geneerisellä työkalulla eteenpäin.

Reply:
Vois olla hyvä idea